src/hamcrest/core/assert_that.py

```diff
@@ -52,14 +53,43 @@ def assert_that(actual, matcher=None, reason=""):
     This is equivalent to the :py:meth:`~unittest.TestCase.assertTrue` method
     of :py:class:`unittest.TestCase`, but offers greater flexibility in test
     writing by being a standalone function.
-
     """
     if isinstance(matcher, Matcher):
-        _assert_match(actual=actual, matcher=matcher, reason=reason)
-    else:
-        if isinstance(actual, Matcher):
-            warnings.warn("arg1 should be boolean, but was {}".format(type(actual)))
-        _assert_bool(assertion=cast(bool, actual), reason=cast(str, matcher))
+        return _assert_match(actual=actual, matcher=matcher, reason=reason)
+
+    # Warn if we got a Matcher first
+    if isinstance(actual, Matcher):
+        warnings.warn("arg1 should be boolean, but was {}".format(type(actual)))
+    # Or if actual and matcher are both truthy strings
+    if isinstance(actual, str) and isinstance(matcher, str) and actual:
+        warnings.warn("assert_that(<str>, <str>) only evaluates whether the "
```

Comment:
Perhaps we could suggest resolving the ambiguity by passing the reason as a named parameter, in the event that they truly intended to do a truthiness assertion? `assert_that(<str>, reason=<str>)`